src/instructlab/data/list.py

```diff
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+from typing import List, Tuple, TypedDict
+import logging
+import time
+
+# Third Party
+import click
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetListing(TypedDict):
+    """
+    Represents a single dataset listing.
+    """
+
+    filename: str
+    created_at: str
+    file_size: str
+
+
+# convert_bytes_to_proper_mag takes a dir/file size in Bytes and converts it to the proper Magnitude (MB, GB)
+def convert_bytes_to_proper_mag(f_size: int) -> Tuple[float, str]:
+    magnitudes = ["KB", "MB", "GB"]
+    magnitude = "B"
+    adjusted_fsize = float(f_size)
+    for mag in magnitudes:
+        if adjusted_fsize >= 1024:
+            magnitude = mag
+            adjusted_fsize /= 1024
+        else:
+            return adjusted_fsize, magnitude
+    return adjusted_fsize, magnitude
+
+
+# print_table displays the datasets on the system in a nicely formatted table
+def print_table(headers: list[str], data: list[list[str]]):
+    column_widths = [
+        max(len(str(row[i])) for row in data + [headers]) for i in range(len(headers))
+    ]
+
+    # Print separator line between headers and data
+    horizontal_lines = ["-" * (width + 2) for width in column_widths]
+    joining_line = "+" + "+".join(horizontal_lines) + "+"
+
+    print(joining_line)
+
+    outputs = []
+    for header, width in zip(headers, column_widths, strict=False):
+        outputs.append(f" {header:{width}} ")
+    print("|" + "|".join(outputs) + "|")
+
+    print(joining_line)
+
+    for row in data:
+        outputs = []
+        for item, width in zip(row, column_widths, strict=False):
+            outputs.append(f" {item:{width}} ")
+        print("|" + "|".join(outputs) + "|")
+
+    print(joining_line)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--dataset-dirs",
+    help="Base directories where datasets are stored.",
+    multiple=True,
+    default=[DEFAULTS.DATASETS_DIR],
+    show_default=True,
+)
+def list_data(dataset_dirs: list[str]):
+    """lists datasets"""
+    entries: List[DatasetListing] = []
+    for directory in dataset_dirs:
+        for entry in Path(directory).iterdir():
+            stat = entry.stat()
+            created_at = time.strftime(
+                "%Y-%m-%d %H:%M:%S", time.localtime(stat.st_ctime)
+            )
+            fsize, mag = convert_bytes_to_proper_mag(stat.st_size)
+            formatted_size = f"{fsize:.2f} {mag}"
```

Comment:
The "bytes-to-string-that-represents-human-readable-size" is useful in both `data` and `model` paths. It should be a util function.

Reply:
This is proposed as part of https://github.com/instructlab/instructlab/pull/1885